style_guide.md

```diff
@@ -0,0 +1,118 @@
+# Style Guide
+
+## What is this?
+This is a set of guidelines on how to *style* code you plan on contributing to the prog-o-meter repository. You don't have to follow each, or any of them, but it may take your PR longer to get merged if someone else has to rewrite and document your code.
+
+## Why is this useful? 
+When everybody is on the same page, everyone writes better code. The purpose of the prog-o-meter project is too get newcomers out of their shell and excited to start learning, and feel confident contributing to the open source community. The faster everyone starts getting comfortable the better the project will be.
+
+## 1. Quick Start
+These are some quick guidelines and principles that this project follows. 
+
+In general...
+* **Structure** should follow the [PEP 8](https://www.python.org/dev/peps/pep-0008/#introduction) format.
+* **Docstrings** should follow the [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) format. (Sec. 3)
+* Code and comments should be kept **beginner friendly**. Purpose and diction should be easy to understand. (Sec. 4)
+* When in doubt look around; you should ***STAY CONSISTENT*** with the rest of the project. 
+
+In order to keep things beginner friendly... 
+* All classes, methods, and functions should include a docstring. (Sec. 3)
+* Loop and control structures should be documented at each line. (Sec. 4)
+
+## 2. Names
+
+### Best Naming Practices
+Names should be **descriptive**, and **concise**.
+*  **Descriptive**: Names should accurately describe the entity they represent. 
+*  **Concise**: Names should be as short as possible without extraneous abbreviation, or loss of meaning. 
+
+"Function" in this section refers to a functions, methods, and generators as group.
+
+### Names of Structures and Data
+Names should follow [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Naming#Naming) convention. Simply put...
+* **Classes** should be upper camel case, e.x. ```class MySuperCoolClass():```
+* **Functions**: should be lowercase underscored, e.x. ```def my_function():``` 
+* **Variables**: should be lowercase underscored, e.x. ```my_integer```
+
+## 3. Comments
+Some of our guidelines are contradictory to [PEP 8](https://www.python.org/dev/peps/pep-0008/#introduction) and [Google Python](http://google.github.io/styleguide/pyguide.html?showone=Comments#Comments) standards.
+We deviate from these formats in an effort to remain **beginner friendly**.
+
+### Inline Comments
+You should document all loop and control structures with inline comments. 
+These comments should be indented at least 8 spaces after the end of the line. 
+Indentation of the comment should match indentation of the line it describes. 
+There should be a space after each '#', and the first letter of the comment should be captialized. 
+Ex. 
+```python
```

Comment:
It is really great you have added examples, but maybe it would be even better if we used examples from our actual codebase?

Reply:
I tried to find an example from the current project, but I couldn't find any that included a nested structure, and followed PEP 8. I agree that an example from the current project would be better, but there are none that meet both of those criteria which I think is essential to get the *best* example.

Reply:
I don't think a nested structure is necessary to get the point across (especially since most of the code contributors will be writing won't have a nested structure). I think giving three seperate examples, a for-statement, a while statement and a if/else (with potential elif's) from the code will be a good way to exemplify here.